angular/projects/spark-angular/src/lib/components/sprk-tabs/sprk-tabs.component.ts

```diff
@@ -0,0 +1,362 @@
+        return (
+          panel.ref.nativeElement.id ===
+          $event.target.getAttribute('aria-controls')
+        );
+      });
+
+      this.resetTabs(
+        this.tabs.map((tab) => tab.ref.nativeElement),
+        this.panels.map((panel) => panel.ref.nativeElement),
+        this.activeClass,
+      );
+
+      this.setActiveTab(
+        $event.target,
+        activePanel.ref.nativeElement,
+        this.activeClass,
+      );
+    }
+  }
+
+  /**
+   * @ignore
+   */
+  @HostListener('window:resize')
+  onResize() {
+    this.ariaOrientation(window.innerWidth, this.ref.nativeElement);
+  }
+
+  /**
+   * @ignore
+   */
+  @HostListener('keydown', ['$event'])
+  onKeydown($event) {
+    const isTabsButton = $event.target.classList.contains(
+      'sprk-c-Tabs__button',
+    );
+    if (!isTabsButton) {
+      return;
+    }
+
+    const keys = {
+      end: 35,
+      home: 36,
+      left: 37,
+      up: 38,
+      right: 39,
+      down: 40,
+      tab: 9,
+    };
+
+    const tabElements = this.tabs.map((tab) => tab.ref.nativeElement);
+    const panelElements = this.panels.map((panel) => panel.ref.nativeElement);
+
+    if ($event.keyCode === keys.left || $event.keyCode === keys.up) {
+      this.incrementTab(tabElements, panelElements, this.activeClass, -1);
+    } else if ($event.keyCode === keys.right || $event.keyCode === keys.down) {
+      this.incrementTab(tabElements, panelElements, this.activeClass, 1);
+    } else if ($event.keyCode === keys.tab) {
+      if ($event.target.getAttribute('role') === 'tab') {
+        $event.preventDefault();
+        panelElements[
+          this.getActiveTabIndex(tabElements, this.activeClass)
+        ].focus();
+      }
+    } else if ($event.keyCode === keys.home) {
+      this.goToEndTab(tabElements, panelElements, this.activeClass, -1);
+    } else if ($event.keyCode === keys.end) {
+      this.goToEndTab(tabElements, panelElements, this.activeClass, 1);
+    }
+  }
+
+  /**
+   * @ignore
+   */
+  getClasses(): string {
+    const classArray: string[] = ['sprk-c-Tabs'];
+
+    if (this.additionalClasses) {
+      this.additionalClasses.split(' ').forEach((className) => {
+        classArray.push(className);
+      });
+    }
+
+    return classArray.join(' ');
+  }
+
+  /**
+   * @ignore
+   */
+  getContentRelationships(): void {
+    let tabIDs = [];
+    let panelIDs = [];
+
+    if (this.tabs && this.panels) {
+      this.tabs.forEach((tab, index) => {
+        const tabID = `tabs-${this.componentID}-tab-${index}`;
+        const panelID = `tabs-${this.componentID}-panel-${index}`;
+
+        this.renderer.setAttribute(tab.ref.nativeElement, 'id', tabID);
+        this.renderer.setAttribute(
+          tab.ref.nativeElement,
+          'aria-controls',
+          panelID,
+        );
+        tabIDs.push(tabID);
+        panelIDs.push(panelID);
+      });
+
+      tabIDs = tabIDs.reverse();
+      panelIDs = panelIDs.reverse();
```

Comment:
Another way to do this is to use splice.
`tabsID.splice(0, 0, tabID)` will get you the same result as push/reverse. Save a few lines of code, but there doesn't seem to be any performance advantage to one or the other. Not a necessary change, just figured I'd throw it out there in case anyone has a preference!